### ISSUE TITLE
Ability to force a data request based on input outside of the table

### ISSUE THREAD

**Issue by roseak** [mbrn/material-table#307]
When I was doing my own server side pagination, data was based on the state's data, but now that the function to fetch the data is called on render, I'm not sure how to toggle a fetch when inputs are submitted outside of the table that can also be used to filter the data.  Suggestions on how to go about this?

**Comment by mbrn** [comment]
I will think about.

**Comment by roseak** [comment]
Not the best of options, but something short term might be having access to triggering a data query based on the toolbar or from free actions.

**Comment by hcsgzh** [comment]
Hi @mbrn ,

I am doing my own server-side filter, sort, and search actions too. If you could open an onQueryChange or handleQueryChange function to triggering a data query based on remote data mode. That will give me the ability to use my own filter UI. For the onQueryChange function, I expect an attribute as a current query object and will return a new query object and will triggering a data query. I am not sure that this is the best way to do. 
What do you think?

**Comment by mbrn** [comment]
Hi @hcsgzh ,

It already call data (think it as onQueryChange) on every change on table. It is expect data from you.

Additionally you want to return query too. Am i right? 

Then i will set your query to table.

**Comment by Daniel0827** [comment]
Hi,

What I did as a temporary solution was to have a component that extends MaterialTable. It has a function that calls onQueryChange(this.state.query) and I pass this function in the constructor to one that's received as a prop. Then I can save the reference of that function outside of the component and I can trigger it manually whenever I want.

This stopped working however since withStyles was added to MaterialTable, so I'm stuck with v1.23.3 for the time being.
